pkg/ds/daemon_set_test.go

```diff
@@ -104,16 +104,17 @@ func watchDSChanges(
 				errCh <- util.Errorf("Error occured when watching daemon set changes: %v", watched.Err)
 			}
 
+			dsID := ds.ID()
```

Comment:
it doesn't fix a race, but theoretically the ID could change between `ID()` calls (although in practice it wouldn't).

I think this just came from me playing with the solution to this race for a while and one incarnation of it didn't involve locking inside of the `ID()` function